src/libraries/System.Linq/src/System/Linq/Distinct.SpeedOpt.cs

```diff
@@ -16,6 +16,8 @@ private sealed partial class DistinctIterator<TSource>
             public override int GetCount(bool onlyIfCheap) => onlyIfCheap ? -1 : new HashSet<TSource>(_source, _comparer).Count;
 
             public override TSource? TryGetFirst(out bool found) => _source.TryGetFirst(out found);
+
+            public override bool Contains(TSource value) => _source.Contains(value);
```

Comment:
`DistinctIterator` may have a custom comparer.

Reply:
Yup, thanks. I'd convinced myself it doesn't matter, but it obviously does.